pkgs/applications/window-managers/cagebreak/default.nix

```diff
@@ -0,0 +1,54 @@
+{ stdenv, fetchFromGitHub
+, meson, ninja, pkg-config, wayland, scdoc, makeWrapper
+, wlroots, wayland-protocols, pixman, libxkbcommon
+, cairo , pango, fontconfig, pandoc, udev
+, xwayland ? null
+, nixosTests
+}:
+
+stdenv.mkDerivation rec {
+  pname = "cagebreak";
+  version = "1.4.2";
+
+  src = fetchFromGitHub {
+    owner = "project-repo";
+    repo = "cagebreak";
+    rev = version;
+    sha256 = "+Ww1rsUR7qe/BixLPR8GiRc3C6QmpLzWpT2wym8b4/M=";
+  };
+
+  nativeBuildInputs = [ meson ninja pkg-config wayland scdoc makeWrapper ];
+
+  buildInputs = [
+    wlroots wayland wayland-protocols pixman libxkbcommon cairo
+    pango.dev fontconfig pandoc udev
+  ];
+
+  outputs = [ "out" "contrib" ];
+
+  mesonFlags = [
+    "-Dxwayland=${stdenv.lib.boolToString (xwayland != null)}"
+    "-Dversion_override=${version}"
+  ];
+
+  mesonBuildType = "release";
+
+  postInstall = ''
+    mkdir -p $contrib/share/cagebreak
+    cp $src/examples/config $contrib/share/cagebreak/config
+  '';
+
+  postFixup = stdenv.lib.optionalString (xwayland != null) ''
+    wrapProgram $out/bin/cagebreak --prefix PATH : "${xwayland}/bin"
+  '';
+
+  passthru.tests.basic = nixosTests.cagebreak;
+
+  meta = with stdenv.lib; {
+    description = "A Wayland tiling compositor inspired by ratpoison";
+    homepage    = "https://github.com/project-repo/cagebreak";
+    license     = licenses.mit;
+    platforms   = platforms.linux;
+    maintainers = with maintainers; [ berbiche ];
```

Comment:
please don't vertically align these. it's a code style that was never suggested in the very minimal section in the manual (and I've been actively trying to purge this)

Reply:
Fixed